bin/propolis-server/src/lib/migrate/source.rs

```diff
@@ -275,25 +452,52 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> SourceProtocol<T> {
 
         let step = bits.len() * 8 * PAGE_SIZE;
         for gpa in (start_gpa..end_gpa).step_by(step) {
+            let mut pages_offered = 0;
             // Always capture the dirty page mask even if the offer discipline
             // says to offer all pages. This ensures that pages that are
-            // transferred now and not touched again will not be offered again.
-            self.track_dirty(GuestAddr(gpa), &mut bits).await?;
+            // transferred now and not touched again will not be offered again
+            // by a subsequent phase.
+            self.track_dirty(GuestAddr(gpa), &mut bits)?;
+
             match offer_discipline {
                 RamOfferDiscipline::OfferAll => {
                     for byte in bits.iter_mut() {
                         *byte = 0xff;
                     }
+                    pages_offered = PAGE_BITMAP_SIZE * 8;
```

Comment:
Is there an edge case here in that, if the region size does not start or end on a page address that is a multiple of 8 pages, this will attempt to send pages that aren't in the region?

Reply:
Well, so, there is a potential edge case, but it won't actually result in us
attempting to send more pages. `pages_offered` doesn't actually _control_ what pages we
offer, instead, we just check if it's non-zero to determine whether to offer
anything. The integer value is only used for logging. So, we may log that we are
offering more pages than we actually are...

Reply:
I believe that `make_mem_offer` already handles the case where the bitmap is
larger than the actual number of pages in the region. So, perhaps the logging
should just happen there, instead.

Reply:
Okee doke.